### PR TITLE
Feat/Iroha2-5 - Add asset details page

### PR DESCRIPTION
**Pull request by VladislavPopovSR**
https://app.zenhub.com/workspaces/iroha-v2-60ddb820813b9100181fc060/issues/zh/5

### CODE REVIEW

File: src/pages/Assets/AssetDetailsPage.vue [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    handleUnknownError(e);
+  } finally {
+    isFetchingAsset.value = false;
+  }
+});
+
+const transactionStatus = ref<ftm.Status>(null);
+const transactionType = ref<TabAssetsSort>('recent');
+
+// use asset's transactions from payload instead or replace with fetching asset's transactions method
```
Comment (0x009922) [inline src/pages/Assets/AssetDetailsPage.vue:58]: I don't understand this comment. Could you rephrase?

And, should it be a TODO/FIXME?

File: src/pages/Assets/AssetDetailsPage.vue [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+      class="asset-details__metrics"
+    >
+      <template #default>
+        <div
+          v-if="isFetchingAsset"
+          class="asset-details__metrics_loading"
+        >
+          <BaseLoading />
+        </div>
+        <div v-else>
```
Comment (0x009922) [inline src/pages/Assets/AssetDetailsPage.vue:75]:
```suggestion
        <div v-else-if="asset">
```

File: src/pages/Assets/AssetDetailsPage.vue [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+          <div class="asset-details__metrics-data">
+            <DataField
+              :title="$t('name')"
+              :value="asset?.id.split('#')[0]"
```
Comment (0x009922) [inline src/pages/Assets/AssetDetailsPage.vue:79]: Split asset id to a name and domain in a `computed`, please.

File: src/shared/lib/localization/en.json
Comment (0x009922) [inline src/shared/lib/localization/en.json:1]: Current structure of the localisation file looks messy and not well-organised. Perhaps will need to revisit in future.

File: src/shared/ui/components/BaseTable.vue [hunk 4]
```diff
@@ -115,8 +114,8 @@ const { width } = useWindowSize();
 const PAGINATION_BREAKPOINT = 960;
 
 const items = computed(() => {
-  if (props.loading && props.pagination) {
-    return Array.from({ length: props.pagination.page_size }, () => null);
+  if (props.loading) {
+    return Array.from({ length: props.pagination?.page_size ?? 10 }, () => null);
   }
```
Comment (0x009922) [inline src/shared/ui/components/BaseTable.vue:119]: I don't immediately like this assumption of 10 by default. Can we leave it as it was for now?

File: src/shared/lib/localization/en.json
Comment (VladislavPopovSR) [reply src/shared/lib/localization/en.json:1]: I removed unused messages and improved a little bit messages' organization.

File: src/shared/ui/components/BaseTable.vue (hunk 4 above)
Comment (VladislavPopovSR) [reply src/shared/ui/components/BaseTable.vue:119]: Fixed✅

File: src/shared/ui/components/BaseTable.vue [hunk 5]
```diff
@@ -115,8 +114,8 @@ const { width } = useWindowSize();
 const PAGINATION_BREAKPOINT = 960;
 
 const items = computed(() => {
-  if (props.loading && props.pagination) {
-    return Array.from({ length: props.pagination.page_size }, () => null);
+  if (props.loading) {
+    return [null];
```
Comment (0x009922) [inline src/shared/ui/components/BaseTable.vue:118]: Why this change at all?

File: src/pages/Accounts/AccountDetailsPage.vue [hunk 6]
```diff
@@ -54,13 +54,15 @@ onMounted(async () => {
   }
 });
 
-// use account's assets from payload instead or replace with fetching account's assets method
+// TODO: use assets from account payload instead
+//  or replace with fetching account's assets method
```
Comment (0x009922) [inline src/pages/Accounts/AccountDetailsPage.vue:58]:
```suggestion
// FIXME: this loads ALL assets, not only belonging to the account.
```

File: src/pages/Accounts/AccountDetailsPage.vue [hunk 7]
```diff
@@ -54,13 +54,15 @@ onMounted(async () => {
...
 const assetsTable = useTable(http.fetchAssetDefinitions);
```
Comment (0x009922) [inline src/pages/Accounts/AccountDetailsPage.vue:59]: Asset definitions do not belong to accounts anyway. They belong to their domains.

Asset = asset definition + account.

```suggestion
const assetsTable = useTable(http.fetchAssets);
```

File: src/pages/Accounts/AccountDetailsPage.vue [hunk 8]
```diff
@@ -54,13 +54,15 @@ onMounted(async () => {
...
 
 const transactionStatus = ref<ftm.Status>(null);
 const transactionType = ref<ftm.TabBlocksScreen>('transactions');
 
-// use account's transactions from payload instead or replace with fetching account's transactions method
+// TODO: use transactions from account payload instead
+//  or replace with fetching specific transactions method
```
Comment (0x009922) [inline src/pages/Accounts/AccountDetailsPage.vue:65]:
```suggestion
// FIXME: this loads ALL transactions, not only related to the account
```

File: src/shared/ui/components/BaseTable.vue (hunk 5 above)
Comment (VladislavPopovSR) [reply src/shared/ui/components/BaseTable.vue:118]: In order to get back to previous expression. It doesn't change anything since display is the same it was before.

File: src/pages/Accounts/AccountDetailsPage.vue (hunk 7 above)
Comment (VladislavPopovSR) [reply src/pages/Accounts/AccountDetailsPage.vue]: ✅

File: src/shared/ui/components/BaseTable.vue (hunk 5 above)
Comment (0x009922) [reply src/shared/ui/components/BaseTable.vue:118]: It used to be that if there is a pagination and no data yet, then it will display `page_size` placeholders. Now it ignores pagination and just displays a single placeholder. So, the display is not the same, or am I missing something?

Comment (VladislavPopovSR) [reply src/shared/ui/components/BaseTable.vue:118]: We have the following condition 
```
v-if="item && (width >= props.breakpoint || !$slots['mobile-card'])" 
```
so either it one or 10 length nullish array - anyway we will display only one row containing BaseLoading component.

**Review by 0x009922 — APPROVED** [review]
